src/agent/coverage/src/block.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(target_os = "linux")]
+pub mod linux;
+
+#[cfg(target_os = "windows")]
+pub mod windows;
+
+use std::collections::BTreeMap;
+use std::path::PathBuf;
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct ModuleCov {
+    pub module: PathBuf,
+    pub blocks: BTreeMap<u64, BlockCov>,
+}
+
+impl ModuleCov {
+    pub fn new(module: impl Into<PathBuf>, blocks: impl IntoIterator<Item = (u64, u32)>) -> Self {
+        let module = module.into();
+        let blocks = blocks
+            .into_iter()
+            .map(|(o, sz)| (o, BlockCov::new(o, sz)))
+            .collect();
+
+        Self { module, blocks }
+    }
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct BlockCov {
+    pub offset: u64,
+    pub size: u32,
+    pub count: u32,
```

Comment:
Is this `count` of how many times a single input hits the block, or of how many inputs hit the block at least once?

Reply:
It is meant to support either, depending on context.

Right now, all breakpoints can only be hit once per input, on both platforms, so `count > 1` only when aggregating runs. In this case, for a corpus of `N` runs, `count / N` would tell us what proportion of the corpus covers the block.

If we added support for recording multiple hits per input, then the meaning would be dependent on the aggregation function used when merging (sum, avg, max, etc).

Reply:
Great, do add a comment.

Note that if breakpoint overhead were slow, we might only want to note hit/not hit so we can set breakpoints on unhit blocks only, though that might not be a big win if we're detecting coverage in every loaded module.

Reply:
Yeah, "only break on uncovered" would be a nice mode for a hot path use case. For now I'll just make sure the meaning is clear in the comment.